lib/renderers/cairo.h

```diff
@@ -247,7 +247,15 @@ bm_cairo_paint(struct cairo *cairo, uint32_t width, uint32_t max_height, const s
     paint.cursor = menu->cursor;
     paint.pos = (struct pos){ (menu->title ? 2 : 0) + result.x_advance, vpadding };
     paint.box = (struct box){ (menu->title ? 2 : 4), 0, vpadding, vpadding, width - paint.pos.x, ascii_height };
-    bm_cairo_draw_line(cairo, &paint, &result, "%s", (menu->filter ? menu->filter : ""));
+
+    char *filter_text = (menu->filter ? menu->filter : "");
+    size_t filter_len = strlen(filter_text);
+    if (menu->password) {
+        memset(filter_text, '*', filter_len);
+        filter_text[filter_len] = '\0';
+    }
+
+    bm_cairo_draw_line(cairo, &paint, &result, "%s", filter_text);
```

Comment:
Instead if memsetting the filter_text, you could instead just draw empty string here if password is true.

Reply:
hmm I pushed an update doing it like that. It's nicer for simplicity but feels/looks less nice I think... I'll leave it up to you